contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol

```diff
@@ -0,0 +1,333 @@
+        );
+        assertEq(
+            startingPSMUnderlyingBalance + amountStableIn,
+            endingPSMUnderlyingBalance
+        );
+    }
+
+    /// @notice pcv deposit gets depleted on redeem
+    function testSwapVoltForUSDC() public {
+        uint256 startingUserUnderlyingBalance = underlyingToken.balanceOf(
+            address(this)
+        );
+        uint256 startingPSMUnderlyingBalance = underlyingToken.balanceOf(
+            address(psm)
+        );
+        uint256 redeemAmountOut = psm.getRedeemAmountOut(mintAmount);
+        uint256 startingUserVOLTBalance = volt.balanceOf(address(this));
+
+        volt.approve(address(psm), mintAmount);
+        uint256 amountOut = psm.redeem(
+            address(this),
+            mintAmount,
+            redeemAmountOut
+        );
+
+        uint256 endingUserVOLTBalance = volt.balanceOf(address(this));
+        uint256 endingUserUnderlyingBalance = underlyingToken.balanceOf(
+            address(this)
+        );
+        uint256 endingPSMUnderlyingBalance = underlyingToken.balanceOf(
+            address(psm)
+        );
+
+        assertEq(startingUserVOLTBalance, endingUserVOLTBalance + mintAmount);
+        assertEq(
+            endingUserUnderlyingBalance,
+            startingUserUnderlyingBalance + amountOut
+        );
+        assertEq(
+            endingPSMUnderlyingBalance,
+            startingPSMUnderlyingBalance - amountOut
+        );
+    }
+
+    /// @notice redeem fails without approval
+    function testSwapVoltForUSDCFailsWithoutApproval() public {
+        vm.expectRevert(bytes("ERC20: transfer amount exceeds allowance"));
+
+        psm.redeem(address(this), mintAmount, mintAmount / 1e12);
+    }
+
+    /// @notice mint fails without approval
+    function testSwapUnderlyingForVoltFailsWithoutApproval() public {
+        vm.expectRevert(bytes("ERC20: transfer amount exceeds allowance"));
+
+        psm.mint(address(this), mintAmount, 0);
+    }
+
+    /// @notice withdraw erc20 fails without correct permissions
+    function testERC20WithdrawFailure() public {
+        vm.expectRevert(bytes("CoreRef: Caller is not a PCV controller"));
+
+        psm.withdrawERC20(address(underlyingToken), address(this), 100);
+    }
+
+    /// @notice withdraw erc20 succeeds with correct permissions
+    function testERC20WithdrawSuccess() public {
+        vm.prank(addresses.voltGovernorAddress);
+        core.grantPCVController(address(this));
+
+        uint256 startingBalance = underlyingToken.balanceOf(address(this));
+        psm.withdrawERC20(address(underlyingToken), address(this), mintAmount);
+        uint256 endingBalance = underlyingToken.balanceOf(address(this));
+
+        assertEq(endingBalance - startingBalance, mintAmount);
+    }
+
+    /// @notice set global rate limited minter fails when caller is governor and new address is 0
+    function testSetPCVDepositFailureZeroAddress() public {
+        vm.startPrank(addresses.voltGovernorAddress);
```

Comment:
`prank` on the `setSurplusTarget` call or does it need to wrap the except revert?

Reply:
I think it needs to wrap the expectRevert, but it works either way. next time I'm writing a test like this I can try .prank vs .startPrank